components/brave_vpn/resources/panel/components/main-panel/index.tsx

```diff
@@ -1,20 +1,36 @@
 import * as React from 'react'
-import * as S from './style'
-import StatusIndicator from '../../components/status-indicator'
-import PanelBox from '../../components/panel-box'
-import Toggle from '../../components/toggle'
 import { SettingsAdvancedIcon, CaratStrongRightIcon } from 'brave-ui/components/icons'
-import { ConnectionState } from '../../api/panel_browser_api'
-import { Region } from '../../api/region_interface'
-interface Props {
-  isOn: boolean
-  status: ConnectionState
-  region: Region
-  onToggleClick: () => void
-  onSelectRegionButtonClick: () => void
-}
 
-function MainPanel (props: Props) {
+import * as S from './style'
+import SelectRegionList from '../select-region-list'
+import PanelBox from '../panel-box'
+import Toggle from '../toggle'
+import ErrorPanel from '../error-panel'
+import { useSelector, useDispatch } from '../../state/hooks'
+import * as Actions from '../../state/actions'
+
+function MainPanel () {
+  const dispatch = useDispatch()
+  const currentRegion = useSelector(state => state.currentRegion)
+  const hasError = useSelector(state => state.hasError)
+  const isSelectingRegion = useSelector(state => state.isSelectingRegion)
+
+  const onSelectRegionButtonClick = () => {
+    dispatch(Actions.isSelectingRegion(true))
+  }
+
+  React.useEffect(() => {
+    dispatch(Actions.initialize())
```

Comment:
Since this UI is the only UI that will show, I wonder if we send it before React even kicks in. It only needs the store, so I don't think there's any reason to wait for React's first render (of nothing). So perhaps dispatch `Actions.initialize()` right from `store.ts`?

Reply:
The api calls inside `initialize` action will only be successful if the user has a valid purchase of the vpn product. Calling it earlier than the main panel can yield into failed calls.

The main panel will only render if only the purchase state is `PURCHASED` - Haven't integrated this flow yet. Will do it on another PR.

Reply:
I still think this logic should be 100% contained in the store or other events, not the UI components. So some handler would handle the purchase state changing, and call the data fetch for initialise accordingly. Sounds like those api calls shouldn't be in initialise then either but should be in "purchaseConfirmed" for example